plugins/AtlasExplorer/api/usecases/OutlineShell.cpp

```diff
@@ -20,100 +20,47 @@
 
 #include "OutlineShell.h"
 
+#include <brayns/engine/common/MathTypesOsprayTraits.h>
 #include <brayns/engine/components/Geometries.h>
+#include <brayns/engine/components/GeometryViews.h>
 #include <brayns/engine/geometry/types/Isosurface.h>
 #include <brayns/engine/systems/GenericBoundsSystem.h>
 #include <brayns/engine/systems/GeometryCommitSystem.h>
 #include <brayns/engine/systems/GeometryInitSystem.h>
 #include <brayns/engine/volume/types/RegularVolume.h>
 
-#include <api/usecases/common/DataUtils.h>
-
 namespace
 {
 class ValidVoxelGridFilter
 {
 public:
-    static std::vector<uint8_t> filter(size_t voxelSize, const std::vector<double> &data)
-    {
-        if (voxelSize == 1)
-        {
-            return _filterScalar(data);
-        }
-
-        return _filterNDimensional(voxelSize, data);
-    }
-
-private:
-    static std::vector<uint8_t> _filterScalar(const std::vector<double> &data)
+    static std::vector<uint8_t> filter(const Atlas &atlas)
     {
-        const auto minMax = DataMinMax::compute(data);
-        const auto minValue = minMax.first;
+        auto voxelCount = atlas.getVoxelCount();
+        auto result = std::vector<uint8_t>(voxelCount, 0u);
 
-        std::vector<uint8_t> result(data.size(), 0u);
-#pragma omp parallel for
-        for (size_t i = 0; i < data.size(); ++i)
+        for (size_t i = 0; i < voxelCount; ++i)
         {
-            auto isEmpty = data[i] == minValue;
-
-            if (!std::isfinite(data[i]) || isEmpty)
+            if (atlas.isValidVoxel(i))
             {
-                continue;
+                result[i] = 255u;
```

Comment:
255u is uint32_t no ? Could we use std::numeric_limits<uint8> or ~0 ?

Reply:
Will change for numeric limits